eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -532,8 +532,11 @@ object Validation {
             handleChannelUpdate(d1, db, nodeParams.currentBlockHeight, Left(lcu))
         }
       case None =>
-        // should never happen, we log a warning and handle the update, it will be rejected since there is no related channel
-        log.warning("unrecognized local channel update for channelId={} localAlias={}", lcu.channelId, lcu.shortIds.localAlias)
+        lcu.shortIds.real.toOption match {
+          case Some(realScid) if d.prunedChannels.contains(realScid) => log.debug("created local channel update for pruned channelId={} scid={}", lcu.channelId, realScid)
+          case _ => log.warning("unrecognized local channel update for channelId={} localAlias={}", lcu.channelId, lcu.shortIds.localAlias)
```

Comment:
I use the "should never happen" when an occurence is definitely a bug, but that's a personal convention.

Reply:
Sounds good, done in https://github.com/ACINQ/eclair/pull/2514/commits/c6e0cec473a6847d7891824794f5deecdc9aca26